source/fe/mapping_fe.cc

```diff
@@ -909,12 +909,82 @@ MappingFE<dim, spacedim>::transform_real_to_unit_cell(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell,
   const Point<spacedim> &                                     p) const
 {
-  Assert(false, StandardExceptions::ExcNotImplemented());
+  const std::vector<Point<spacedim>> support_points =
+    this->compute_mapping_support_points(cell);
+
+  Tensor<1, spacedim> DF[dim];
+  // [TODO]
+  // When 2nd derivatives are implemented, we'll uncomment this
+  // Tensor<1, spacedim> D2F[dim][dim];
+
+  Point<dim>      p_unit;
+  Point<spacedim> mapped_point;
+  Point<spacedim> p_minus_F;
+
+  Point<dim>     f;
+  Tensor<2, dim> df;
+
+  const double       eps        = 1.e-12 * cell->diameter();
+  const unsigned int loop_limit = 10;
+
+  unsigned int loop = 0;
+  do
+    {
+      auto mapped_point = support_points[0] * 0.0;
+      for (unsigned int i = 0; i < this->fe->n_dofs_per_cell(); ++i)
+        {
+          mapped_point += support_points[i] * this->fe->shape_value(i, p_unit);
+          auto F = this->fe->shape_grad(i, p_unit);
+          // [TODO]
+          // When 2nd derivatives are implemented, we'll uncomment this
+          // auto H = this->fe->shape_grad_grad(i, p_unit);
+          for (unsigned int j = 0; j < dim; ++j)
+            {
+              DF[j] += F[j] * support_points[i];
+              // [TODO]
+              // When 2nd derivatives are implemented, we'll uncomment this
+              // for (unsigned int l = 0; l < dim; ++l)
+              //   D2F[j][l] += H[j][l] * support_points[i];
+            }
+        }
+
+      p_minus_F = p;
+      p_minus_F -= mapped_point;
+
+      for (unsigned int j = 0; j < dim; ++j)
+        {
+          f[j] = DF[j] * p_minus_F;
+          for (unsigned int l = 0; l < dim; ++l)
+            // [TODO]
+            // When 2nd derivatives are implemented, we'll uncomment this
+            // df[j][l] = -DF[j] * DF[l] + D2F[j][l] * p_minus_F;
+            df[j][l] = -DF[j] * DF[l];
+        }
+      // Solve  [df(x)]d=f(x)
+      const Tensor<1, dim> d =
+        invert(df) * static_cast<const Tensor<1, dim> &>(f);
+      p_unit -= d;
+
+      // Clear DF and D2F
+      for (unsigned int j = 0; j < dim; ++j)
+        {
+          DF[j].clear();
```

Comment:
You only need this because you re-use the variable in the next loop iteration and you declared it outside the loop. Just move the declaration of the variable into the loop -- then it dies at the `}` and is initialized automatically once you start the next iteration.